torchmdnet/models/model.py

```diff
@@ -380,9 +387,14 @@ def forward(
 
         if self.derivative:
             pos.requires_grad_(True)
+                   
         # run the potentially wrapped representation model
         x, v, z, pos, batch = self.representation_model(
-            z, pos, batch, box=box, q=q, s=s
+            z,
+            pos,
+            batch,
+            box=box,
+            extra_args = extra_args if self.representation_model.additional_methods is not None else None,
```

Comment:
You are assuming all representation models have a member called "additional_methods", which I do not think is a good idea. This is an internal implementation detail being leaked.
Why not sending extra_args all the time to the model? It is free to ignore them.

Reply:
Each model has `additional_methods` defined in its initialization. I was thinking about it at some point, but then I need in any case a way in the forward method of each architecture to do some assertion and pre-processing on the data. So at the end, I decided to use as common line `if extra_args is not None:` in the forward of the architecture (see [this ](https://github.com/torchmd/torchmd-net/blob/2b18f905f51c118ac1bcf30119bc6d02e4fd0669/torchmdnet/models/tensornet.py#L261)for example)

Reply:
My point is architectures should not require to have a member called `additional_methods`. And external classes should not depend on it or access it in general.
Both parties (TorchMD_Net and the representation models) have enough information on their own. 
TorchMD_Net should just send extra_args all the time, you are assuming extra_args will be used exclusively for your new additional labels feature, but this might not be the case at all. I could implement it so that I can send in extra_args something like "check_errors", for instance.
OTOH the representation models have several members that they can use to know what to use from extra_args, if anything.

Reply:
ending extra_args all the time to the model according also to the other comments in this PR